torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -359,11 +355,11 @@ def with_comms(func: TestFunc) -> TestFunc:
     def wrapper(
         self, *args: Tuple[object], **kwargs: Dict[str, Any]  # type: ignore[misc]
     ) -> None:
-        # if backend not specified, and cuda available, then use nccl, else gloo
-        if torch.cuda.is_available() and torch.cuda.device_count() >= self.world_size:
-            self.device_type = "cuda"
-        else:
+        # if enough GPU we can use GPU, otherwise we fallback to CPU
+        if torch.cuda.is_available() and torch.cuda.device_count() < self.world_size:
             self.device_type = "cpu"
```

Comment:
I understand your intention, but the "and" here seems a bit weird. 
What about:
```
if not torch.cuda.is_available() or torch.cuda.device_count() < self.world_size:
```
Or maybe I missed your other intention.

Reply:
Sure that works too, let me change to that

Reply:
this might also fix the current CI issues, let's see